### DISCUSSION TITLE
Chromebook

### DISCUSSION THREAD

**Discussion by brhodewalt** [pybricks/support#391] [Q&A] [answered]
I'm a teacher of Intro to CS. I have one EV3, and I hope to get more. Each student has a Chromebook. I can't tell if this software will run on Chrome. If it doesn't, is there another Python that will? 

Our Intro to CS course focuses on Python. I would really like to add robots to the current curriculum. (We do have occasional access to Macs, but I would prefer student do their Python coding on their 24/7 devices.)

Thanks in advance for any help!

**Comment by laurensvalk** [comment, marked as answer]
One thing to keep in mind when investing in new materials, is that EV3 is being discontinued. Depending on what you need, you may be interested in LEGO SPIKE Prime as well.

Pybricks for EV3 requires Visual Studio Code. It appears that Visual Studio Code can run on Chromebooks with some effort, but I'm not sure about the EV3 MicroPython extension. Since you have an EV3, perhaps you can try this out?

For all the other hubs, including SPIKE Prime which we will support soon, all you need is Bluetooth and a browser.

**Reply by brhodewalt** [reply]
I'm happy to give this a shot. Can you tell me where I can find the EV3 MicroPython extension? Thanks.

**Reply by laurensvalk** [reply]
You can follow the instructions for installation and running programs here: https://pybricks.com/ev3-micropython/

**Reply by brhodewalt** [reply]
It does appear that VSC will run on Chromebooks, but not on mine. Ours are managed by our school district, and installing Java (required for VSC) is not an option. Thanks for trying.

**Reply by laurensvalk** [reply]
Ah, yes, IT restrictions can always be a bit cumbersome, so that's another thing to keep in mind. 

That's another reason why we're currently focusing on a browser based coding solution for the modern programmable LEGO hubs, including SPIKE Prime and others.

**Reply by TechnoX** [reply]
@laurensvalk It's not technically possible to send program to EV3 over bluetooth from the browser? What is the limitation and why does it work for other hubs? :) 
I understand that it isn't implemented right now, but will it be able to implement?

**Reply by laurensvalk** [reply]
One key difference is that the newer hubs support Bluetooth Low Energy (BLE), which is what we currently use in the browser. The EV3 has only Bluetooth ("Bluetooth Classic").

That isn't to say that EV3 support is impossible, though: There is also web serial, where the browser can access (existing) serial ports. This could mean that some tools might still have to be installed on the user computer to configure everything.

Perhaps more importantly, it would be great to find out if there is still interest for continued EV3 support in Pybricks. That is, whether we should continue supporting EV3 beyond version 2.0 (which is currently used by LEGO education.)

**Comment by brhodewalt** [comment]
This is very helpful. Thank you.

Bruce Rhodewalt (he/him/his)
RMHS Math Teacher
Google Educator
***@***.***
http://rhodewalt.com/room608.html


________________________________________
